docs/side_quests/metadata.md

```diff
@@ -0,0 +1,732 @@
+    ├── cowpy-hola.txt
+    └── cowpy-salut.txt
+```
+
+Success! All our phrases are correctly sorted and we now see which of them correspond to which language.
+
+Let's take a look at `cowpy-salut.txt`:
+
+```console title="cowpy-salut.txt"
+ ____________________
+/ Salut, ça va?      \
+\ à plus dans le bus /
+ --------------------
+\                             .       .
+ \                           / `.   .' "
+  \                  .---.  <    > <    >  .---.
+   \                 |    \  \ - ~ ~ - /  /    |
+         _____          ..-~             ~-..-~
+        |     |   \~~~\.'                    `./~~~/
+       ---------   \__/                        \__/
+      .'  O    \     /               /       \  "
+     (_____,    `._.'               |         }  \/~~~/
+      `----.          /       }     |        /    \__/
+            `-.      |       /      |       /      `. ,~~|
+                ~-.__|      /_ - ~ ^|      /- _      `..-'
+                     |     /        |     /     ~-.     `-. _  _  _
+                     |_____|        |_____|         ~ - . _ _ _ _ _>
+```
+
+Look through the other files. All phrases should be spoken by the fashionable stegosaurus.
+
+How did this work? The `publishDir` directive is evaluated at runtime when the process executes. Each process task gets its own meta map from the input tuple When the directive is evaluated, `${meta.lang_group}` is replaced with the actual group language value for that sample creating the dynamic paths like `results/romance`.
+
+### 3.2. Customize the character
+
+In our samplesheet, we have another column: `character`. To tailor the tool parameters per sample, we can also access information from the `meta` map in the script section. This is really useful in cases were a tool should have different parameters for each sample.
+
+Let's customize the characters by changing the `cowpy` command:
+
+=== "After"
+
+    ```groovy title="main.nf" linenums="37" hl_lines="3"
+    script:
+    """
+    cat $input_file | cowpy -c ${meta.character} > cowpy-${input_file}
+    """
+    ```
+
+=== "Before"
+
+    ```groovy title="main.nf" linenums="37"
+    script:
+    """
+    cat $input_file | cowpy -c "stegosaurus" > cowpy-${input_file}
+    """
+    ```
+
+Let's run this:
+
+```bash title="Use cowpy"
+nextflow run main.nf -resume
+```
+
+and take another look at our french phrase:
+
+```console title="romance/cowpy-salut.txt"
+ ____________________
+/ Salut, ça va?      \
+\ à plus dans le bus /
+ --------------------
+  \
+   \   \_\_    _/_/
+    \      \__/
+           (oo)\_______
+           (__)\       )\/\
+               ||----w |
+               ||     ||
+```
+
+This is a subtle difference to other parameters that we have set in the pipelines in previous trainings. A parameter that is passed as part of the `params` object is generally applied to all samples. When a more surgical approache is necessary, using the sample specific information is a good alternative.
```

Comment:
```suggestion

```

I'm not sure I understand the association with `params`.

Reply:
what I meant to say was something like: params typically applied to all files, not tailored to a specific file. Using meta data as shown here allows us to fine tune things per file.

I'll try rephrasing